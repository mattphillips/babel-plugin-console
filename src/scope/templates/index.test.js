import each from 'jest-each';

import templateHelpers from './';

describe('Templates', () => {
  const noop = () => {};

  describe('.buildGroupCollapsed', () => {
    each([[0], [1], [2], [3]]).it('calls template with correct string and indentation level: %d', level => {
      const template = jest.fn(() => noop);
      const { buildGroupCollapsed } = templateHelpers(template);
      buildGroupCollapsed('bob')(level);
      expect(template.mock.calls[0][0]).toMatchSnapshot();
    });

    it('invokes template with given title as an object', () => {
      const template = () => templateCurry;
      const templateCurry = jest.fn();
      const { buildGroupCollapsed } = templateHelpers(template);
      buildGroupCollapsed('matt')(0);
      expect(templateCurry).toHaveBeenCalledWith({ TITLE: 'matt' });
    });

    it('returns value from invoking template', () => {
      const template = () => () => 'hello world';
      const { buildGroupCollapsed } = templateHelpers(template);
      expect(buildGroupCollapsed('matt', 'woz', 'ere')(0)).toBe('hello world');
    });
  });

  describe('.buildGroupEnd', () => {
    it('calls template with correct string', () => {
      const template = jest.fn(() => noop);
      const { buildGroupEnd } = templateHelpers(template);
      buildGroupEnd('bob');
      expect(template.mock.calls[0][0]).toMatchSnapshot();
    });

    it('invokes template with given title as an object', () => {
      const template = () => templateCurry;
      const templateCurry = jest.fn();
      const { buildGroupEnd } = templateHelpers(template);
      buildGroupEnd('matt');
      expect(templateCurry).toHaveBeenCalledWith({ TITLE: 'matt' });
    });

    it('returns value from invoking template', () => {
      const template = () => () => 'hello world';
      const { buildGroupEnd } = templateHelpers(template);
      expect(buildGroupEnd('matt', 'woz', 'ere')).toBe('hello world');
    });
  });

  describe('.buildLog', () => {
    it('calls template with console log string when indentation level is 0', () => {
      const template = jest.fn(() => noop);
      const { buildLog } = templateHelpers(template);
      buildLog('bob', 'woz', 'ere')(0);
      expect(template).toHaveBeenCalledWith('console.log(ARGS);');
    });

    each([[1], [2], [3]]).it('calls template with correct string and indentation level: %d', level => {
      const template = jest.fn(() => noop);
      const { buildLog } = templateHelpers(template);
      buildLog('bob', 'woz', 'ere')(level);
      expect(template.mock.calls[0][0]).toMatchSnapshot();
    });

    it('invokes template with given args as an object', () => {
      const template = () => templateCurry;
      const templateCurry = jest.fn();
      const { buildLog } = templateHelpers(template);
      buildLog('matt', 'woz', 'ere')(0);
      expect(templateCurry).toHaveBeenCalledWith({ ARGS: ['matt', 'woz', 'ere'] });
    });

    it('returns value from invoking template', () => {
      const template = () => () => 'hello world';
      const { buildLog } = templateHelpers(template);
      expect(buildLog('matt', 'woz', 'ere')(0)).toBe('hello world');
    });
  });
});
