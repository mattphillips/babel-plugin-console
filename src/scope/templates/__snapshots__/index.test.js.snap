// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Templates .buildGroupCollapsed calls template with correct string and indentation level: 0 1`] = `
"
    if (typeof window !== 'undefined') {
      console.groupCollapsed(TITLE);
    } else {
      console.log('▼ ', TITLE);
    }
  "
`;

exports[`Templates .buildGroupCollapsed calls template with correct string and indentation level: 1 1`] = `
"
    if (typeof window !== 'undefined') {
      console.groupCollapsed(TITLE);
    } else {
      console.log('| ▼ ', TITLE);
    }
  "
`;

exports[`Templates .buildGroupCollapsed calls template with correct string and indentation level: 2 1`] = `
"
    if (typeof window !== 'undefined') {
      console.groupCollapsed(TITLE);
    } else {
      console.log('| | ▼ ', TITLE);
    }
  "
`;

exports[`Templates .buildGroupCollapsed calls template with correct string and indentation level: 3 1`] = `
"
    if (typeof window !== 'undefined') {
      console.groupCollapsed(TITLE);
    } else {
      console.log('| | | ▼ ', TITLE);
    }
  "
`;

exports[`Templates .buildGroupEnd calls template with correct string 1`] = `
"
  if (typeof window !== 'undefined') {
    console.groupEnd(TITLE);
  }
"
`;

exports[`Templates .buildLog calls template with correct string and indentation level: 1 1`] = `
"
    if (typeof window !== 'undefined') {
      console.log(ARGS);
    } else {
      console.log('| | ', ARGS);

    }
  "
`;

exports[`Templates .buildLog calls template with correct string and indentation level: 2 1`] = `
"
    if (typeof window !== 'undefined') {
      console.log(ARGS);
    } else {
      console.log('| | | ', ARGS);

    }
  "
`;

exports[`Templates .buildLog calls template with correct string and indentation level: 3 1`] = `
"
    if (typeof window !== 'undefined') {
      console.log(ARGS);
    } else {
      console.log('| | | | ', ARGS);

    }
  "
`;
